queries.ghactivity.php

```diff
@@ -494,4 +479,135 @@ public static function get_monitored_repos( $fields = 'id=>name' ) {
 
 		return $repos_to_monitor;
 	}
+
+	/**
+	 * Walking through label meta and filters out labels from other repos, from closed issues, and unlabeled labels
+	 * Returns an array of array of label slugs and passed time since it was labeled
+	 *
+	 * @since 2.0.0
+	 *
+	 * @param array  $meta Array of label meta objects.
+	 * @param string $repo_name name of the repo.
+	 *
+	 * @return array
+	 */
+	public static function filter_labeled_labels( $meta, $repo_name ) {
+		$dates = array();
+		$slugs = array();
+		foreach ( $meta as $repo_slug => $serialized ) {
+			// count only issues from specific repo.
+			if ( strpos( strtolower( $repo_slug ), strtolower( $repo_name ) ) === 0 ) {
+				$issue_number = explode( '#', $repo_slug )[1];
+				$post_id      = self::find_open_gh_issue( $repo_name, $issue_number );
+				$label_ary    = unserialize( $serialized[0] );
+
+				// We want to capture only opened, labeled issues.
+				if ( $post_id && 'labeled' === $label_ary['status'] ) {
+					$time                = time() - strtotime( $label_ary['labeled'] );
+					$dates[]             = $time;
+					$slugs[ $repo_slug ] = $time;
+				}
+			}
+		}
+
+		return array( $slugs, $dates );
+	}
+
+	/**
+	 * Returns associated array of repo labels and issue slugs labeled with these labels
+	 *
+	 * @since 2.0.0
+	 *
+	 * @param string $repo_name name of the repo.
+	 *
+	 * @return array
```

Comment:
I think it may be useful to give more details about the expected array here, as a kind of example. It could help us when coming back to that code in a few months.